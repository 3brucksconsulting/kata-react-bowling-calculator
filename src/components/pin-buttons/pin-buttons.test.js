import React from 'react';
import ReactTestRender from 'react-test-renderer';
import PinButtons from './pin-buttons';

describe('The PinButtons component', () => {
  it('givenOneAvailablePin_rendersAsExpected', () => {
    const nextAvailablePins = 1;
    const isGameOver = false;

    const component = ReactTestRender.create(
      <PinButtons
        nextAvailablePins={nextAvailablePins}
        isGameOver={isGameOver}
      />
    );

    let result = component.toJSON();

    expect(result).toMatchSnapshot();
  });

  it('givenTwoAvailablePins_rendersAsExpected', () => {
    const nextAvailablePins = 2;
    const isGameOver = false;

    const component = ReactTestRender.create(
      <PinButtons
        nextAvailablePins={nextAvailablePins}
        isGameOver={isGameOver}
      />
    );

    let result = component.toJSON();

    expect(result).toMatchSnapshot();
  });

  it('givenThreeAvailablePins_rendersAsExpected', () => {
    const nextAvailablePins = 3;
    const isGameOver = false;

    const component = ReactTestRender.create(
      <PinButtons
        nextAvailablePins={nextAvailablePins}
        isGameOver={isGameOver}
      />
    );

    let result = component.toJSON();

    expect(result).toMatchSnapshot();
  });

  it('givenFourAvailablePins_rendersAsExpected', () => {
    const nextAvailablePins = 4;
    const isGameOver = false;

    const component = ReactTestRender.create(
      <PinButtons
        nextAvailablePins={nextAvailablePins}
        isGameOver={isGameOver}
      />
    );

    let result = component.toJSON();

    expect(result).toMatchSnapshot();
  });

  it('givenFiveAvailablePins_rendersAsExpected', () => {
    const nextAvailablePins = 5;
    const isGameOver = false;

    const component = ReactTestRender.create(
      <PinButtons
        nextAvailablePins={nextAvailablePins}
        isGameOver={isGameOver}
      />
    );

    let result = component.toJSON();

    expect(result).toMatchSnapshot();
  });

  it('givenSixAvailablePins_rendersAsExpected', () => {
    const nextAvailablePins = 6;
    const isGameOver = false;

    const component = ReactTestRender.create(
      <PinButtons
        nextAvailablePins={nextAvailablePins}
        isGameOver={isGameOver}
      />
    );

    let result = component.toJSON();

    expect(result).toMatchSnapshot();
  });

  it('givenSevenAvailablePins_rendersAsExpected', () => {
    const nextAvailablePins = 7;
    const isGameOver = false;

    const component = ReactTestRender.create(
      <PinButtons
        nextAvailablePins={nextAvailablePins}
        isGameOver={isGameOver}
      />
    );

    let result = component.toJSON();

    expect(result).toMatchSnapshot();
  });

  it('givenEightAvailablePins_rendersAsExpected', () => {
    const nextAvailablePins = 8;
    const isGameOver = false;

    const component = ReactTestRender.create(
      <PinButtons
        nextAvailablePins={nextAvailablePins}
        isGameOver={isGameOver}
      />
    );

    let result = component.toJSON();

    expect(result).toMatchSnapshot();
  });

  it('givenNineAvailablePins_rendersAsExpected', () => {
    const nextAvailablePins = 9;
    const isGameOver = false;

    const component = ReactTestRender.create(
      <PinButtons
        nextAvailablePins={nextAvailablePins}
        isGameOver={isGameOver}
      />
    );

    let result = component.toJSON();

    expect(result).toMatchSnapshot();
  });

  it('givenTenAvailablePins_rendersAsExpected', () => {
    const nextAvailablePins = 10;
    const isGameOver = false;

    const component = ReactTestRender.create(
      <PinButtons
        nextAvailablePins={nextAvailablePins}
        isGameOver={isGameOver}
      />
    );

    let result = component.toJSON();

    expect(result).toMatchSnapshot();
  });

  it('givenIsGameOverTrue_rendersAsExpected', () => {
    const nextAvailablePins = 10;
    const isGameOver = true;

    const component = ReactTestRender.create(
      <PinButtons
        nextAvailablePins={nextAvailablePins}
        isGameOver={isGameOver}
      />
    );

    let result = component.toJSON();

    expect(result).toMatchSnapshot();
  });
});
