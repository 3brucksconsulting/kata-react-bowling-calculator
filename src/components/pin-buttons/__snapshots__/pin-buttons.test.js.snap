// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The PinButtons component givenEightAvailablePins_rendersAsExpected 1`] = `
<div
  className="row"
>
  <div
    className="col-md-9"
  >
    <h5>
      Click the Number of Pins Knocked Down
    </h5>
    <div
      className="btn-toolbar"
    >
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={0}
        type="button"
        value={0}
      >
        0
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={1}
        type="button"
        value={1}
      >
        1
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={2}
        type="button"
        value={2}
      >
        2
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={3}
        type="button"
        value={3}
      >
        3
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={4}
        type="button"
        value={4}
      >
        4
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={5}
        type="button"
        value={5}
      >
        5
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={6}
        type="button"
        value={6}
      >
        6
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={7}
        type="button"
        value={7}
      >
        7
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={8}
        type="button"
        value={8}
      >
        8
      </button>
    </div>
  </div>
  <div
    className="col-md-3"
  >
    <button
      className="btn btn-primary"
      disabled={true}
      title="Start New Game"
      type="button"
    >
      Start New Game
    </button>
  </div>
</div>
`;

exports[`The PinButtons component givenFiveAvailablePins_rendersAsExpected 1`] = `
<div
  className="row"
>
  <div
    className="col-md-9"
  >
    <h5>
      Click the Number of Pins Knocked Down
    </h5>
    <div
      className="btn-toolbar"
    >
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={0}
        type="button"
        value={0}
      >
        0
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={1}
        type="button"
        value={1}
      >
        1
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={2}
        type="button"
        value={2}
      >
        2
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={3}
        type="button"
        value={3}
      >
        3
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={4}
        type="button"
        value={4}
      >
        4
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={5}
        type="button"
        value={5}
      >
        5
      </button>
    </div>
  </div>
  <div
    className="col-md-3"
  >
    <button
      className="btn btn-primary"
      disabled={true}
      title="Start New Game"
      type="button"
    >
      Start New Game
    </button>
  </div>
</div>
`;

exports[`The PinButtons component givenFourAvailablePins_rendersAsExpected 1`] = `
<div
  className="row"
>
  <div
    className="col-md-9"
  >
    <h5>
      Click the Number of Pins Knocked Down
    </h5>
    <div
      className="btn-toolbar"
    >
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={0}
        type="button"
        value={0}
      >
        0
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={1}
        type="button"
        value={1}
      >
        1
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={2}
        type="button"
        value={2}
      >
        2
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={3}
        type="button"
        value={3}
      >
        3
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={4}
        type="button"
        value={4}
      >
        4
      </button>
    </div>
  </div>
  <div
    className="col-md-3"
  >
    <button
      className="btn btn-primary"
      disabled={true}
      title="Start New Game"
      type="button"
    >
      Start New Game
    </button>
  </div>
</div>
`;

exports[`The PinButtons component givenIsGameOverTrue_rendersAsExpected 1`] = `
<div
  className="row"
>
  <div
    className="col-md-9"
  >
    <h5>
      Click the Number of Pins Knocked Down
    </h5>
    <div
      className="btn-toolbar"
    >
      <button
        className="btn btn-outline-dark"
        disabled={true}
        title={0}
        type="button"
        value={0}
      >
        0
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={true}
        title={1}
        type="button"
        value={1}
      >
        1
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={true}
        title={2}
        type="button"
        value={2}
      >
        2
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={true}
        title={3}
        type="button"
        value={3}
      >
        3
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={true}
        title={4}
        type="button"
        value={4}
      >
        4
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={true}
        title={5}
        type="button"
        value={5}
      >
        5
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={true}
        title={6}
        type="button"
        value={6}
      >
        6
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={true}
        title={7}
        type="button"
        value={7}
      >
        7
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={true}
        title={8}
        type="button"
        value={8}
      >
        8
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={true}
        title={9}
        type="button"
        value={9}
      >
        9
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={true}
        title={10}
        type="button"
        value={10}
      >
        10
      </button>
    </div>
  </div>
  <div
    className="col-md-3"
  >
    <button
      className="btn btn-primary"
      disabled={false}
      title="Start New Game"
      type="button"
    >
      Start New Game
    </button>
  </div>
</div>
`;

exports[`The PinButtons component givenNineAvailablePins_rendersAsExpected 1`] = `
<div
  className="row"
>
  <div
    className="col-md-9"
  >
    <h5>
      Click the Number of Pins Knocked Down
    </h5>
    <div
      className="btn-toolbar"
    >
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={0}
        type="button"
        value={0}
      >
        0
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={1}
        type="button"
        value={1}
      >
        1
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={2}
        type="button"
        value={2}
      >
        2
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={3}
        type="button"
        value={3}
      >
        3
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={4}
        type="button"
        value={4}
      >
        4
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={5}
        type="button"
        value={5}
      >
        5
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={6}
        type="button"
        value={6}
      >
        6
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={7}
        type="button"
        value={7}
      >
        7
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={8}
        type="button"
        value={8}
      >
        8
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={9}
        type="button"
        value={9}
      >
        9
      </button>
    </div>
  </div>
  <div
    className="col-md-3"
  >
    <button
      className="btn btn-primary"
      disabled={true}
      title="Start New Game"
      type="button"
    >
      Start New Game
    </button>
  </div>
</div>
`;

exports[`The PinButtons component givenOneAvailablePin_rendersAsExpected 1`] = `
<div
  className="row"
>
  <div
    className="col-md-9"
  >
    <h5>
      Click the Number of Pins Knocked Down
    </h5>
    <div
      className="btn-toolbar"
    >
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={0}
        type="button"
        value={0}
      >
        0
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={1}
        type="button"
        value={1}
      >
        1
      </button>
    </div>
  </div>
  <div
    className="col-md-3"
  >
    <button
      className="btn btn-primary"
      disabled={true}
      title="Start New Game"
      type="button"
    >
      Start New Game
    </button>
  </div>
</div>
`;

exports[`The PinButtons component givenSevenAvailablePins_rendersAsExpected 1`] = `
<div
  className="row"
>
  <div
    className="col-md-9"
  >
    <h5>
      Click the Number of Pins Knocked Down
    </h5>
    <div
      className="btn-toolbar"
    >
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={0}
        type="button"
        value={0}
      >
        0
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={1}
        type="button"
        value={1}
      >
        1
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={2}
        type="button"
        value={2}
      >
        2
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={3}
        type="button"
        value={3}
      >
        3
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={4}
        type="button"
        value={4}
      >
        4
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={5}
        type="button"
        value={5}
      >
        5
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={6}
        type="button"
        value={6}
      >
        6
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={7}
        type="button"
        value={7}
      >
        7
      </button>
    </div>
  </div>
  <div
    className="col-md-3"
  >
    <button
      className="btn btn-primary"
      disabled={true}
      title="Start New Game"
      type="button"
    >
      Start New Game
    </button>
  </div>
</div>
`;

exports[`The PinButtons component givenSixAvailablePins_rendersAsExpected 1`] = `
<div
  className="row"
>
  <div
    className="col-md-9"
  >
    <h5>
      Click the Number of Pins Knocked Down
    </h5>
    <div
      className="btn-toolbar"
    >
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={0}
        type="button"
        value={0}
      >
        0
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={1}
        type="button"
        value={1}
      >
        1
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={2}
        type="button"
        value={2}
      >
        2
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={3}
        type="button"
        value={3}
      >
        3
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={4}
        type="button"
        value={4}
      >
        4
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={5}
        type="button"
        value={5}
      >
        5
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={6}
        type="button"
        value={6}
      >
        6
      </button>
    </div>
  </div>
  <div
    className="col-md-3"
  >
    <button
      className="btn btn-primary"
      disabled={true}
      title="Start New Game"
      type="button"
    >
      Start New Game
    </button>
  </div>
</div>
`;

exports[`The PinButtons component givenTenAvailablePins_rendersAsExpected 1`] = `
<div
  className="row"
>
  <div
    className="col-md-9"
  >
    <h5>
      Click the Number of Pins Knocked Down
    </h5>
    <div
      className="btn-toolbar"
    >
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={0}
        type="button"
        value={0}
      >
        0
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={1}
        type="button"
        value={1}
      >
        1
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={2}
        type="button"
        value={2}
      >
        2
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={3}
        type="button"
        value={3}
      >
        3
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={4}
        type="button"
        value={4}
      >
        4
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={5}
        type="button"
        value={5}
      >
        5
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={6}
        type="button"
        value={6}
      >
        6
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={7}
        type="button"
        value={7}
      >
        7
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={8}
        type="button"
        value={8}
      >
        8
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={9}
        type="button"
        value={9}
      >
        9
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={10}
        type="button"
        value={10}
      >
        10
      </button>
    </div>
  </div>
  <div
    className="col-md-3"
  >
    <button
      className="btn btn-primary"
      disabled={true}
      title="Start New Game"
      type="button"
    >
      Start New Game
    </button>
  </div>
</div>
`;

exports[`The PinButtons component givenThreeAvailablePins_rendersAsExpected 1`] = `
<div
  className="row"
>
  <div
    className="col-md-9"
  >
    <h5>
      Click the Number of Pins Knocked Down
    </h5>
    <div
      className="btn-toolbar"
    >
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={0}
        type="button"
        value={0}
      >
        0
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={1}
        type="button"
        value={1}
      >
        1
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={2}
        type="button"
        value={2}
      >
        2
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={3}
        type="button"
        value={3}
      >
        3
      </button>
    </div>
  </div>
  <div
    className="col-md-3"
  >
    <button
      className="btn btn-primary"
      disabled={true}
      title="Start New Game"
      type="button"
    >
      Start New Game
    </button>
  </div>
</div>
`;

exports[`The PinButtons component givenTwoAvailablePins_rendersAsExpected 1`] = `
<div
  className="row"
>
  <div
    className="col-md-9"
  >
    <h5>
      Click the Number of Pins Knocked Down
    </h5>
    <div
      className="btn-toolbar"
    >
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={0}
        type="button"
        value={0}
      >
        0
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={1}
        type="button"
        value={1}
      >
        1
      </button>
      <button
        className="btn btn-outline-dark"
        disabled={false}
        title={2}
        type="button"
        value={2}
      >
        2
      </button>
    </div>
  </div>
  <div
    className="col-md-3"
  >
    <button
      className="btn btn-primary"
      disabled={true}
      title="Start New Game"
      type="button"
    >
      Start New Game
    </button>
  </div>
</div>
`;
